### ISSUE TITLE
Unhandled rejection Error: semverutils.parseRange returned null when trying to parse ""

### ISSUE THREAD

**Issue by XhmikosR** [tjunnone/npm-check-updates#220]
```
C:\Users\xmr\Desktop\bootstrap>node -v && npm -v
v5.9.0
3.7.3

C:\Users\xmr\Desktop\bootstrap>ncu --version
2.5.8

C:\Users\xmr\Desktop\bootstrap>git status
On branch v4-dev
Your branch is up-to-date with 'origin/v4-dev'.
nothing to commit, working directory clean

C:\Users\xmr\Desktop\bootstrap>ncu
Unhandled rejection Error: semverutils.parseRange returned null when trying to parse "". This is probably a problem with the "semver-utils" dependency. Please report an issue at https://github.com/tjunnone/npm-check-updates/issues.
    at Object.numParts (C:\Users\xmr\AppData\Roaming\npm\node_modules\npm-check-updates\lib\version-util.js:28:15)
    at C:\Users\xmr\AppData\Roaming\npm\node_modules\npm-check-updates\lib\versionmanager.js:52:32
    at C:\Users\xmr\AppData\Roaming\npm\node_modules\npm-check-updates\node_modules\lodash\index.js:7113:30
    at C:\Users\xmr\AppData\Roaming\npm\node_modules\npm-check-updates\node_modules\lodash\index.js:2275:27
    at C:\Users\xmr\AppData\Roaming\npm\node_modules\npm-check-updates\node_modules\lodash\index.js:3049:15
    at baseMap (C:\Users\xmr\AppData\Roaming\npm\node_modules\npm-check-updates\node_modules\lodash\index.js:2274:7)
    at Function.sortBy (C:\Users\xmr\AppData\Roaming\npm\node_modules\npm-check-updates\node_modules\lodash\index.js:7112:20)
    at baseWrapperValue (C:\Users\xmr\AppData\Roaming\npm\node_modules\npm-check-updates\node_modules\lodash\index.js:2768:30)
    at LodashWrapper.wrapperValue (C:\Users\xmr\AppData\Roaming\npm\node_modules\npm-check-updates\node_modules\lodash\index.js:6124:14)
    at upgradeDependencyDeclaration (C:\Users\xmr\AppData\Roaming\npm\node_modules\npm-check-updates\lib\versionmanager.js:54:10)
    at wrapper (C:\Users\xmr\AppData\Roaming\npm\node_modules\npm-check-updates\node_modules\lodash\index.js:3533:19)
    at C:\Users\xmr\AppData\Roaming\npm\node_modules\npm-check-updates\node_modules\lodash\index.js:8201:21
    at C:\Users\xmr\AppData\Roaming\npm\node_modules\npm-check-updates\node_modules\lodash\index.js:3395:24
    at C:\Users\xmr\AppData\Roaming\npm\node_modules\npm-check-updates\node_modules\lodash\index.js:3073:15
    at baseForOwn (C:\Users\xmr\AppData\Roaming\npm\node_modules\npm-check-updates\node_modules\lodash\index.js:2046:14)
    at Function.mapValues (C:\Users\xmr\AppData\Roaming\npm\node_modules\npm-check-updates\node_modules\lodash\index.js:3394:9)
```

Doesn't happen in our master branch, only in the v4-dev.

**Comment by XhmikosR** [comment]
OK, I've isolated this to this line

```json
  "dependencies": {
    "jquery": "1.9.1 - 2"
  }
```

**Comment by cvrebert** [comment]
Presumably https://github.com/coolaj86/semver-utils is buggy in not supporting the dashed range syntax.

**Comment by metaraine** [comment]
Thanks for tracking that down. I successfully submitted some pull requests
to semver-utils a while back to handle a few broken edge cases and add unit
tests. Not sure when I have time to work on this next, but open to pull
requests.

**Comment by metaraine** [comment]
Actually, semver-utils [does support dashed range syntax](https://github.com/coolaj86/semver-utils/blob/master/test/spec.js#L160-L174). (I should know, I [wrote the test spec](https://github.com/coolaj86/semver-utils/blame/master/test/spec.js#L160-L174)!)

Your error says its getting an empty string: `semverutils.parseRange returned null when trying to parse ""`. I would be curious where it is getting an empty string from. That seems to be the issue.

**Comment by XhmikosR** [comment]
@metaraine: just check out the v4-dev branch and see for yourself.

This is a clear bug, not a question. And if I remove the jquery dep, the error goes away.

**Comment by metaraine** [comment]
Okay, thanks. I will investigate further.

**Comment by metaraine** [comment]
Found it, thanks! Published to v2.6.1.

Please be aware that the expected behavior of ncu when upgrading ranges is to replace the whole range with the latest version. The semantics of how the range could be conserved when the latest version is completely outside the range is unclear. I would assume that ^ would be used in normal cases, and a range would only be used if there was some kind of change that broke semver. If you need to exclude jquery from ncu while updating to preserve this range, you can use the new reject syntax: `ncu -x jquery`.

**Comment by XhmikosR** [comment]
Thanks for the fix!

I guess it's OK for now since no error is thrown.
